lxd/resources/pci.go

```diff
@@ -0,0 +1,106 @@
+package resources
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/jaypipes/pcidb"
+	"github.com/pkg/errors"
+
+	"github.com/lxc/lxd/shared/api"
+)
+
+// GetPCI returns a filled api.ResourcesPCI struct ready for use by LXD
+func GetPCI() (*api.ResourcesPCI, error) {
+	pci := api.ResourcesPCI{}
+
+	// Load PCI database
+	pciDB, err := pcidb.New()
+	if err != nil {
+		pciDB = nil
+	}
+
+	// List all PCi devices
+	entries, err := ioutil.ReadDir(sysBusPci)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to list %q", sysBusPci)
+	}
+
+	for _, entry := range entries {
+		entryPath := filepath.Join(sysBusPci, entry.Name())
+		device := api.ResourcesPCIDevice{}
+
+		// Get driver name
+		driverPath := filepath.Join(entryPath, "driver")
+
+		if sysfsExists(driverPath) {
+			target, err := os.Readlink(driverPath)
```

Comment:
Again, please do it the same way we've done so far, use `linkTarget` and `EvalSymlinks`.